file/src/main/scala/akka/stream/alpakka/file/internal/ChangeableFileSubscriber.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.file.internal
+
+import java.nio.channels.FileChannel
+import java.nio.file.{Path, StandardOpenOption}
+
+import akka.Done
+import akka.actor.{ActorLogging, Deploy, Props}
+import akka.stream.IOResult
+import akka.stream.actor.{ActorSubscriberMessage, WatermarkRequestStrategy}
+import akka.stream.alpakka.file.internal.ChangeableFileSubscriber.{ByteStringWrapper, ChangePathMessage}
+import akka.util.ByteString
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Promise
+import scala.util.{Failure, Success}
+
+protected[akka] object ChangeableFileSubscriber {
+  def props(completionPromise: Promise[IOResult], bufSize: Int, openOptions: Set[StandardOpenOption]) = {
+    require(bufSize > 0, "buffer size must be > 0")
+    Props(new ChangeableFileSubscriber(completionPromise, bufSize, openOptions)).withDeploy(Deploy.local)
+  }
+
+  class ChangeableFileSubscriberMessages
+
+  case class ByteStringWrapper(bytes: ByteString) extends ChangeableFileSubscriberMessages
+
+  case class ChangePathMessage(path: Path) extends ChangeableFileSubscriberMessages
+
+}
+
+protected[akka] class ChangeableFileSubscriber(
+    completionPromise: Promise[IOResult],
+    bufSize: Int,
+    openOptions: Set[StandardOpenOption]
+) extends akka.stream.actor.ActorSubscriber
```

Comment:
`ActorSubscriber` will be deprecated in 2.5.0, all that it can do can be done with `GraphStage` instead, so use that. There is plenty of samples in Alpakka to look at for inspiration already.